singer_sdk/sinks/core.py

```diff
@@ -132,7 +148,83 @@ def is_full(self) -> bool:
         Returns:
             True if the sink needs to be drained.
         """
-        return self.current_size >= self.max_size
+        return (
+            self.current_size >= self.sink_timer.sink_max_size
+            if self.batch_dynamic_management and self.sink_timer is not None
+            else self.current_size >= self.max_size
+        )
+
+    @property
```

Comment:
I don't think this needs to be a property, wdyt?

Reply:
I think it makes sense as a property.  I like the cleanness of being able to access it via `.is_too_old`.  That keeps it consistent with `.is_full`.  It is a computed value.  I don't need to pass any parameter to obtain it.  I get a single value back which is used in a comparison when processing each record. If anything I am thinking maybe change it to `@cached_property` since it will never change so why compute it over and over again.  wdyt?

Reply:
Oh, I think `@cached_property` definitely would not work since the value is definitely dynamic.

I agree that even for the sake of consistency with `.is_full` it makes sense as a property 👍

Reply:
Yep after I hit comment I realize how dumb that was.